src/proto/Ethereum.proto

```diff
@@ -74,39 +74,46 @@ message Transaction {
     }
 }
 
+enum TransactionMode {
+    Legacy = 0; // Legacy transaction, pre-EIP2718/EIP1559; for fee gasPrice/gasLimit is used
+    Enveloped = 1; // Enveloped transaction EIP2718 (with type 0x2), fee is according to EIP1559 (base fee, inclusion fee, ...) 
+}
+
 // Input data necessary to create a signed transaction.
-// Supported:
-// - Legacy transaction, with legacy fee (pre-EIP1559; gas price, gas limit)
-// - EIP1559 fee (transaction type: ?; max inclusion fee, max fee per gas)
+// Legacy and EIP2718/EIP1559 transactions supported, see TransactionMode.
 message SigningInput {
     // Chain identifier (256-bit number)
     bytes chain_id = 1;
 
     // Nonce (256-bit number)
     bytes nonce = 2;
 
+    // Transaction version selector: Legacy or enveloped, has impact on fee structure.
+    // Default is Legacy (value 0)
+    TransactionMode tx_mode = 3;
```

Comment:
since you are changing proto message indexes, it will break backward compatibility. Should we just use `enum` for type of message? as an example:

```
enum Type {
   case regular (gasPrice, gasLimit)
   case eip1559(max_inclusion_fee_per_gas, max_fee_per_gas)
} 
```

Reply:
The transaction type and the fee structure are independent dimensions.
We have currently 6 transaction types, so we had to duplicate to 12.  Quite some code duplication.

Reply:
@hewigovens how we gonna handle this level of transaction construction with WalletConnect integration?

Reply:
I think we can add a nested message to represent fee, `enum` type is a bit limited in proto